public/js/actions/sources.js

```diff
@@ -12,17 +12,35 @@ const { getSource, getSourceText } = require("../selectors");
 const constants = require("../constants");
 const Prefs = require("../prefs");
 
-// delay is in ms
-const FETCH_SOURCE_RESPONSE_DELAY = 200;
+/**
+ * Throttles source dispatching to reduce rendering churn.
+ */
+let newSources = [];
+let newSourceTimeout;
+function queueSourcesDispatch(dispatch) {
+  if(!newSourceTimeout) {
+    // Even just batching every 10ms works because we just want to
+    // group sources that come in at once.
+    newSourceTimeout = setTimeout(() => {
+      dispatch({ type: constants.ADD_SOURCES,
+                 sources: newSources });
+      newSources = [];
+      newSourceTimeout = null;
+    }, 10);
+  }
+}
```

Comment:
It might be nice to put a simple debounce function in util and use that

```js
const newSources = debounce(function(dispatch) {
  dispatch({ 
    type: constants.ADD_SOURCES,
    sources: newSourcesList
  });
  newSourcesList = [];
}); 
```